angular-workspace/projects/ni/nimble-angular/src/directives/theme-provider/tests/nimble-theme-provider.directive.spec.ts

```diff
@@ -11,4 +13,199 @@ describe('Nimble theme provider', () => {
+            @ViewChild('themeProvider', { read: ElementRef }) public elementRef: ElementRef<ThemeProvider>;
+        }
+
+        let fixture: ComponentFixture<TestHostComponent>;
+        let directive: NimbleThemeProviderDirective;
+        let nativeElement: ThemeProvider;
+
+        beforeEach(() => {
+            TestBed.configureTestingModule({
+                declarations: [TestHostComponent],
+                imports: [NimbleThemeProviderModule]
+            });
+            fixture = TestBed.createComponent(TestHostComponent);
+            fixture.detectChanges();
+            directive = fixture.componentInstance.directive;
+            nativeElement = fixture.componentInstance.elementRef.nativeElement;
+        });
+
+        it('will use template string values for theme', () => {
+            expect(directive.theme).toBe(Theme.dark);
+            expect(nativeElement.theme).toBe(Theme.dark);
+        });
+
+        it('will use template string values for lang', () => {
+            expect(directive.lang).toBe('de-DE');
+            expect(nativeElement.lang).toBe('de-DE');
+        });
+    });
+
+    describe('with property bound values', () => {
+        @Component({
+            template: `
+                <nimble-theme-provider #themeProvider
+                    [theme]="theme"
+                    [lang]="lang">
+                </nimble-theme-provider>
+            `
+        })
+        class TestHostComponent {
+            @ViewChild('themeProvider', { read: NimbleThemeProviderDirective }) public directive: NimbleThemeProviderDirective;
+            @ViewChild('themeProvider', { read: ElementRef }) public elementRef: ElementRef<ThemeProvider>;
+            public theme: Theme = Theme.dark;
+            public lang = 'de-DE';
+        }
+
+        let fixture: ComponentFixture<TestHostComponent>;
+        let directive: NimbleThemeProviderDirective;
+        let nativeElement: ThemeProvider;
+
+        beforeEach(() => {
+            TestBed.configureTestingModule({
+                declarations: [TestHostComponent],
+                imports: [NimbleThemeProviderModule]
+            });
+            fixture = TestBed.createComponent(TestHostComponent);
+            fixture.detectChanges();
+            directive = fixture.componentInstance.directive;
+            nativeElement = fixture.componentInstance.elementRef.nativeElement;
+        });
+
+        it('can be configured with property binding for theme', () => {
+            expect(directive.theme).toBe(Theme.dark);
+            expect(nativeElement.theme).toBe(Theme.dark);
+
+            fixture.componentInstance.theme = Theme.color;
+            fixture.detectChanges();
+
+            expect(directive.theme).toBe(Theme.color);
+            expect(nativeElement.theme).toBe(Theme.color);
+        });
+
+        it('can be configured with property binding for lang', () => {
```

Comment:
Right now, this test is testing two different things:
1. can use property binding for lang
2. updating lang to an invalid value results in the element to be invalid

I think that would be better separate it into two different tests. To follow the patterns we already have in place, the "can be configured with property binding" should only test the binding aspect of the directive. My thought is that test should set a valid new value for 'lang'. Then, you could have a different test that setting an invalid value for 'lang' appropriately makes the element invalid.